omf-python/src/grid.rs

```diff
@@ -0,0 +1,144 @@
+use omf::{Grid2, Orient2, Vector3};
+use pyo3::prelude::*;
+use pyo3_stub_gen::derive::*;
+
+use crate::array::PyScalarArray;
+
+#[gen_stub_pyclass]
+#[pyclass(name = "Regular")]
+#[derive(Clone)]
+/// Regularly spaced cells.
+pub struct PyGrid2Regular(Grid2);
+
+impl TryFrom<Grid2> for PyGrid2Regular {
+    type Error = ();
+
+    fn try_from(value: Grid2) -> Result<Self, Self::Error> {
+        match value {
+            // A Regular grid can be converted to PyGrid2Regular.
+            Grid2::Regular { .. } => Ok(Self(value)),
+            _ => Err(()),
```

Comment:
I think a panicking `From` implementation would be generally unexpected. [The `From` documentation](https://doc.rust-lang.org/std/convert/trait.From.html#when-to-implement-from) also recommends against implementations that panic.

Reply:
In a typical situation I would agree, but I feel like this isn't one. If it was, using `()` as the error type probably wouldn't be great either.

For self-contained code where the only way it can go wrong is if that code is wrong, a panic is often more clear. Here it seems like everything that calls this immediately chains to `.expect(...)` and it would be more obvious to just panic right away and avoid the extra code.

I don't feel too strongly about it in this specific case, but there was a pattern of unnecessary error handling in other places as well.